test/functional/rpc_preciousblock.py

```diff
@@ -38,26 +38,24 @@ def set_test_params(self):
         self.setup_clean_chain = True
         self.num_nodes = 3
 
-    def skip_test_if_missing_module(self):
-        self.skip_if_no_wallet()
-
     def setup_network(self):
         self.setup_nodes()
 
     def run_test(self):
         self.log.info("Ensure submitblock can in principle reorg to a competing chain")
-        self.nodes[0].generate(1)
+        gen_address = lambda i: self.nodes[i].get_deterministic_priv_key().address  # A non-wallet address to mine to
```

Comment:
nit, in remaining tests this auxiliary function is not used, maybe it could be removed and inline where necessary.

Reply:
There are 5 instances of calling `gen_address` in this test. I think in-lining would make this test overly verbose.

Reply:
IMO it should be consistent, which helps in moving code around, copy&paste and readability. So either expose the auxiliary function or inline everywhere.

Reply:
My rationale was make a clean looking code. Wherever there were many occurrences for the call to `get_deterministic_priv_key()`, I used a variable (in 1 test case where there was only 1 node) and auxiliary function in this test(where there are multiple nodes).